### DISCUSSION TITLE
how to get it running

### DISCUSSION THREAD

**Discussion by heiwilOS** [Mogztter/asciidoctor-web-pdf#402] [Q&A]
Hi,
I started with Mint Xfce

```
asciidoctor-web-pdf ./documentation.adoc
```

results:

```
Unable to generate the PDF - Error: Error: Could not find browser revision 818858. Run "PUPPETEER_PRODUCT=firefox npm install" or "PUPPETEER_PRODUCT=firefox yarn install" to download a supported Firefox browser binary.
````

so far so clear, but 

```console
$ sudo PUPPETEER_PRODUCT=firefox npm install
up to date, audited 1 package in 311ms
found 0 vulnerabilities
```

doesn't change anything.

I have installed google-chrome and
tried the command 
```
$ sudo PUPPETEER_PRODUCT=google-chrome npm install
```

without success

What else I have to do or (and) what I did wrong?

Heiko

**Comment by ggrossetie** [comment]
Hi Heiko!

Not sure why Puppeteer is suggesting to install Firefox and not Chromium... :thinking: 
Do you have the install log when you did `npm i asciidoctor-pdf`?

Also I would strongly recommend to install `asciidoctor-pdf` locally if you are not using `nvm`. Global install in the system space has proven to be unreliable. So I prefer to install in the user space using `nvm`.
If you don't have `nvm`, the easiest way is probably to create a npm project:

```
$ mkdir test-asciidoctor-web-pdf; cd test-asciidoctor-web-pdf; npm init -y; npm i @asciidoctor/core asciidoctor-pdf --save-dev
```
Then you can run `asciidoctor-web-pdf` using:

```
$ ./node_modules/.bin/asciidoctor-web-pdf
```

Let me know if it works for you.
By the way, which version of `node` and `npm` are you using? If you are using a recent version, the following commands should also work:

```
$ npx asciidoctor-web-pdf
$ npm x asciidoctor-web-pdf
```

**Reply by heiwilOS** [reply]
H Guillaume,
the install log after doing `npm i asciidoctor-pdf` I don't have anymore.
But your guided way was working, I get pdf files now. 

```
$ mkdir test-asciidoctor-web-pdf; cd test-asciidoctor-web-pdf; npm init -y; npm i @asciidoctor/core asciidoctor-pdf --save-dev
Wrote to /home/heiwil/test-asciidoctor-web-pdf/package.json:

{
  "name": "test-asciidoctor-web-pdf",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}


npm WARN deprecated fsevents@2.1.3: "Please update to latest v2.3 or v2.2"
npm WARN deprecated @babel/polyfill@7.12.1: 🚨 This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (when needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.

added 137 packages, and audited 138 packages in 2m

10 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
Can I ignore the warnings?


here the two version numbers you have asked for.
```
$ node -v
v10.19.0
```

```
4 npm -v
7.5.3
```

Heiko
